app/kuma-dp/pkg/dataplane/envoy/envoy.go

```diff
@@ -38,6 +41,49 @@ type Envoy struct {
 	opts Opts
 }
 
+func getSelfPath() (string, error) {
+	ex, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Dir(ex), nil
+}
+
+func lookupBinaryPath(candidatePaths []string) (string, error) {
+	for _, candidatePath := range candidatePaths {
+		path, err := exec.LookPath(candidatePath)
+		if err == nil {
+			return path, nil
+		}
+	}
+
+	return "", errors.New("could not find binary")
```

Comment:
can we change the message to `.Errorf("could not find Envoy binary at any of the following locations: %v", candidatePaths)` ?